riscv-page-tables/src/page_table.rs

```diff
@@ -737,6 +779,26 @@ impl<T: PagingMode> PageTableInner<T> {
             _ => Err(Error::PageNotInvalidated),
         }
     }
+
+    /// Promotes the table entry at `vaddr` as a valid leaf.
+    fn promote_leaf(
```

Comment:
`promote_pte()`? we're not actually promoting a leaf; we're turning a table into a leaf pte.

Reply:
Agreed.